### PR TITLE
Add pytorch gptq notebook

### PR DESCRIPTION
**Pull request by reuvenperetz**
## Pull Request Description:
Add notebook for GPTQ in pytorch.

## Checklist before requesting a review:
- [ ] I set the appropriate labels on the pull request.
- [ ] I have added/updated the release note draft (if necessary).
- [ ] I have updated the documentation to reflect my changes (if necessary).
- [ ] All function and files are well documented. 
- [ ] All function and classes have type hints.
- [ ] There is a licenses in all file.
- [ ] The function and variable names are informative. 
- [ ] I have checked for code duplications.
- [ ] I have added new unittest (if necessary).

### CODE REVIEW

File: tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+    },
+    {
+      "cell_type": "markdown",
+      "id": "743dbc3d",
+      "metadata": {
+        "id": "743dbc3d"
+      },
+      "source": [
+        "This tutorial demonstrates the process of retraining and quantizing a MobileNetV2 on CIFAR100 dataset. It starts by fine-tuning a pretrained MobileNetV2 model on the CIFAR100 dataset. After retraining, the model is quantized using Gradient-Based PTQ (GPTQ) which uses unlabeled data for fine-tuning and improving the quantized model accuracy. This tutorial specifically uses mixed-precision quantization, which assigns different precision levels to different layers in the model based on their impact on the output. The quantized model is then evaluated and exported to an ONNX file."
+      ]
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb:41]: "the model is quantized using Gradient-Based PTQ (GPTQ) which uses unlabeled data for fine-tuning" - I think it better to remove the "fine tuning" here since it can be confusing. (you mentioned fine tuning in the previous stage...). maybe: "weights adjustment "

File: tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb [hunk 2]
```diff
@@ -0,0 +1,717 @@
+{
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb:1]: Consider renaming the tutorial to mixed_precision_gptq ?

File: tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+      "id": "3ce7789b-aa3d-4a44-8dc5-dc052ece9cad",
+      "metadata": {
+        "id": "3ce7789b-aa3d-4a44-8dc5-dc052ece9cad"
+      },
+      "outputs": [],
+      "source": [
+        "# Create a mixed-precision quantization configuration with possible mixed-precision search options.\n",
+        "# MCT will search a mixed-precision configuration (namely, bit-width for each layer)\n",
+        "# and quantize the model according to this configuration.\n",
+        "# The candidates bit-width for quantization should be defined in the target platform model:\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb:497]: I would make the comments shorter (even remove it) since you already comment in the description above.

File: tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+      "execution_count": null,
+      "id": "ec6f2ea5-9a79-4961-b746-aac45a95aecb",
+      "metadata": {
+        "id": "ec6f2ea5-9a79-4961-b746-aac45a95aecb"
+      },
+      "outputs": [],
+      "source": [
+        "# Get KPI information to constraint your model's memory size.\n",
+        "# Retrieve a KPI object with helpful information of each KPI metric,\n",
+        "# to constraint the quantized model to the desired memory size.\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb:524]: same as here - it might be too much description..

File: tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+        "                                                  representative_data_gen,\n",
+        "                                                  configuration,\n",
+        "                                                  target_platform_capabilities=target_platform_cap)\n",
+        "\n",
+        "# Set a constraint for each of the KPI metrics.\n",
+        "# Create a KPI object to limit our returned model's size. Note that this values affect only layers and attributes\n",
+        "# that should be quantized (for example, the kernel of Conv2D in Pytorch will be affected by this value,\n",
+        "# while the bias will not)\n",
+        "# examples:\n",
+        "# weights_compression_ratio = 0.75 - About 0.75 of the model's weights memory size when quantized with 8 bits.\n",
```
Comment (Idan-BenAmi) [inline tutorials/notebooks/pytorch/gptq/example_pytorch_mobilenetv2_cifar100_gptq.ipynb:535]: same as here - consider to remove/short the comments

**Comment by ofirgo** [comment]
Closed due to a change in the requirements from a GPTQ tutorial.
